CMakeLists.txt

```diff
@@ -22,10 +22,13 @@ set(minZlibVersion 1.2.11)
 include(CheckCXXSourceCompiles) # TODO: migrate to CheckSourceCompiles in CMake >= 3.19
 include(FeatureSummary)
 include(FeatureOptionsSetup)
+
 feature_option(STACKTRACE "Enable stacktraces" ON)
 feature_option(GUI "Build GUI application" ON)
 feature_option(WEBUI "Enables built-in HTTP server for headless use" ON)
 feature_option(VERBOSE_CONFIGURE "Show information about PACKAGES_FOUND and PACKAGES_NOT_FOUND in the configure output (only useful for debugging the CMake build scripts)" OFF)
+feature_option(DEFINE_COMMIT_HASH "Defines git commit hash in version.h, WARNING: may increase incremental build times" OFF)
```

Comment:
I think we can omit the warning.

Reply:
And the description is poor, maybe something like: "Show git commit hash in version strings"?

Reply:
>And the description is poor, maybe something like: "Show git commit hash in version strings"?

I've been saying this for many comments above, but @jagannatharjun  still can't figure it out...
Description (as well as feature name) should make sense to whoever will use it (assuming the user doesn't know in advance what it does).
Why would the user know about some implementation details like that it defines some preprocessor definitions in some specific source files? All the user needs to know is that enabling this option will result in "Showing the commit hash along with the version"! Even the fact that this is Git commit at the current point in history doesn't matter. Tomorrow we can move to a different version control system, but the meaning of the feature will not change from this, we will just need to change the implementation behind the scenes.